manifests/init.pp

```diff
@@ -84,18 +77,18 @@
     fedora: {
       # Fedora distros < 22 are EOL as of 2015-12-01
       # https://fedoraproject.org/wiki/End_of_life
-      if $::operatingsystemmajrelease < '22' {
-        fail("Unsupported OS: ${::operatingsystem} ${::operatingsystemmajrelease}")
+      if $::lsbmajdistrelease < '22' {
+        fail("Unsupported OS: ${::lsbdistid} ${::lsbmajdistrelease}")
       }
     }
     ubuntu: {
-      if $::operatingsystemmajrelease != '14.04' {
-        # Only tested on 14.04
-        fail("Unsupported OS: ${::operatingsystem} ${::operatingsystemmajrelease}")
+      if $::lsbdistrelease < '16.04' {
+        # Only tested on 16.04
+        fail("Unsupported OS: ${::lsbdistid} ${::lsbdistrelease}")
       }
     }
     default: {
-      fail("Unsupported OS: ${::operatingsystem}")
+      fail("Unsupported OS: ${::lsbdistid}")
```

Comment:
This is where the Beaker CentOS 6-testing tests are failing. Note that they are printing null for the var, which supports the idea that it's a problem with change of vars.